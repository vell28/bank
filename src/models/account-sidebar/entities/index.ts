export interface IAccountSidebar {
  [key: string]: boolean;
}
