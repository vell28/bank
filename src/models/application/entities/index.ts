export interface IApplication {
  applicationId: string;
  productId: string;
  Authorization: string;
  deviceId?: string;

  readonly bankName: string;
  readonly BIC: string;
}
