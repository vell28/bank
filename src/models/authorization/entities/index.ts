export interface IAuthorization {
  service_code?: string;
  phone?: string;
  token?: string;
  sessid?: string;
  token_expires?: Date;
  // maybe its not required
  'dev-id'?: string;
  'token-firebase'?: string;
  isShownLogoutConfirmModal: boolean;
}

export interface ITokenRefresh {
  success: boolean;
  token: string;
}

export interface ISuccessLogout {
  success: boolean;
}

export interface ICodeRequest {
  sessid: string;
  code: string | null;
  success?: boolean;
}

export interface IPasswordCheckRequest {
  password: string;
  phone: string;
}

export interface ISuccessCheckResponse {
  status: 'ok';
}
