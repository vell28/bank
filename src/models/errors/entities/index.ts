export type AppErrorType = '401' | '404' | '403' | '500' | 'ERR_INTERNET_DISCONNECTED';

export interface IAppErrors {
  title?: string;
  subtitle?: string;
  onOk?: () => void;
  btnText?: string;
  supportUrl?: string;
  type?: AppErrorType;
  meta?: string;
}
