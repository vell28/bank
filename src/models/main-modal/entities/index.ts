import { AllMainModalContentsType } from 'containers/operations/types';

export type MainModalContentType = AllMainModalContentsType | '';

export interface IMainModalSettings {
  id: MainModalContentType;
  bindUrl: string;
  title?: string;
}

export interface IMainModal {
  isShown: boolean;
  settings: IMainModalSettings;
  step: number;
}
