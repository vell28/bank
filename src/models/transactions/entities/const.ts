export const FILTER_DATE_FORMAT = 'dd/MM/yyyy';
