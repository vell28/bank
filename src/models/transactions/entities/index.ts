import { CurrencyCodeType } from 'modules/currencies';
import { IAccount, AccountType } from '../../organizations/entities';

export type TransactionStatusType =
  | 'PROCESSING'
  | 'COMPLETED'
  | 'BANK_CANCELLED'
  | 'INSUFFICIENT_FUNDS'
  | 'REFUND'
  | 'HOLD'
  | 'UNKNOWN';

export type TransactionPaymentType = 'TRANSFER' | 'CARD_OPERATION' | 'HOLD' | 'CONVERSION' | 'SEPA' | 'SWIFT';

export interface IOpenApi {
  type: string;
  toCard: string | null;
  toPhone: string | null;
  subType: string | null;
  externalOwnerID: string | null;
  fromCardPan: string | null;
}

export interface IRepeatPayment {
  type: string;
  fromPhone: string;
  account: string;
  fromCardPan: string | null;
  fromCardId: string | null;
  beneficiaryName: string | null;
  beneficiaryAccount: string | null;
  country: string | null;
  city: string | null;
  address: string | null;
  swiftCode: string | null;
  beneficiaryBank: string | null;
  intermediaryBank: string | null;
  intermediarySwift: string | null;
  intermediaryAccount: string | null;
  commissionType: string | null;
  urgency: string | null;
  iban: string | null;
  phoneNumber: string | null;
  cardNumber: string | null;
  currencyCode: CurrencyCodeType;
  amount: number;
  purpose: string | null;
  transferDetails: string | null;
}

export interface ITransaction {
  id: string;
  accountId: string;
  amount: number;
  fee: number;
  status: TransactionStatusType;
  paymentType: TransactionPaymentType;
  fromNumber: string;
  fromName: string;
  fromBank: string;
  toNumber: string;
  toName: string;
  toBank: string;
  currency: {
    code: CurrencyCodeType;
  };
  simpleAccount?: IAccount;
  clientId: string;
  description: string;
  executedAt: string;
  referenceNumber: string;
  paymentToRepeat: IRepeatPayment | null;
  openapiData: IOpenApi | null;
  operationType: string | null;
  fromPanDisplay: string | null;
}

export type ITransactionList = ITransaction[];

export interface ITransactionPagination {
  maxPageNumber: number;
  pageSize: number;
  pageNumber: number;
  length: number;
}

export interface IHistoryFilter {
  date?: string;
  accountType?: AccountType;
  currency?: CurrencyCodeType;
  isShown: boolean;
}

export interface IShownBtns {
  [key: string]: boolean;
}

// transaction id -> settings
export interface IHistorySetting {
  shownBtns: IShownBtns;
  receiptId?: string;
  repeatId?: string;
}

export interface ITransactionResponse extends ITransactionPagination {
  content: ITransactionList;
}

export interface IOrderedTransactions {
  [key: string]: ITransactionList;
}
