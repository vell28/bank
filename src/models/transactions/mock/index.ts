import { CurrencyCodes } from 'modules/currencies';
import { ITransaction, ITransactionResponse } from '../entities';

export const transactionData: ITransaction = {
  id: 'PPY184598',
  amount: -10.0,
  fee: 0,
  status: 'COMPLETED',
  paymentType: 'TRANSFER',
  fromNumber: 'MT98PAPY36836000002676370004519',
  fromName: 'Ralf Williams',
  fromBank: 'Papaya',
  toNumber: 'MT97PAPY36836000002676370003470',
  toName: 'Dmitrii Popov',
  toBank: 'Papaya',
  currency: {
    code: CurrencyCodes.EUR,
  },
  clientId: '6449',
  accountId: 'PPY5107',
  description: 'To: Dmitrii Popov Vvg',
  executedAt: '2019-10-11T10:12:00',
  referenceNumber: 'PPY1e1d31e3-6acb-4d60-9f70-b9151221d11',
  paymentToRepeat: {
    type: 'PAYMENT_PHONE',
    fromPhone: '79111111111',
    account: 'PPY5107',
    fromCardPan: null,
    fromCardId: null,
    beneficiaryName: null,
    beneficiaryAccount: null,
    country: null,
    city: null,
    address: null,
    swiftCode: null,
    beneficiaryBank: null,
    intermediaryBank: null,
    intermediarySwift: null,
    intermediaryAccount: null,
    commissionType: null,
    urgency: null,
    iban: null,
    phoneNumber: '79085491121',
    cardNumber: null,
    currencyCode: CurrencyCodes.EUR,
    amount: 10,
    purpose: 'other Vvgh',
    transferDetails: null,
  },
  openapiData: {
    type: 'TO_CONTACT',
    toCard: null,
    toPhone: '79085491121',
    subType: null,
    externalOwnerID: null,
    fromCardPan: null,
  },
  operationType: null,
  fromPanDisplay: null,
};

const second: ITransaction = {
  id: 'PPY184597',
  amount: -1.0,
  fee: 0,
  status: 'COMPLETED',
  paymentType: 'TRANSFER',
  fromNumber: 'MT98PAPY36836000002676370004519',
  fromName: 'Ralf Williams',
  fromBank: 'Papaya',
  toNumber: 'MT52PAPY36836000002676370000303',
  toName: 'Elfur Long',
  toBank: 'Papaya',
  currency: {
    code: CurrencyCodes.EUR,
  },
  clientId: '6449',
  accountId: 'PPY5107',
  description: 'To: Elfur Long 12',
  executedAt: '2019-10-11T10:08:47',
  referenceNumber: 'PPY84f7087c-c511-4928-b299-13ee47c1bd9',
  paymentToRepeat: {
    type: 'PAYMENT_PHONE',
    fromPhone: '79111111111',
    account: 'PPY5107',
    fromCardPan: null,
    fromCardId: null,
    beneficiaryName: null,
    beneficiaryAccount: null,
    country: null,
    city: null,
    address: null,
    swiftCode: null,
    beneficiaryBank: null,
    intermediaryBank: null,
    intermediarySwift: null,
    intermediaryAccount: null,
    commissionType: null,
    urgency: null,
    iban: null,
    phoneNumber: '79032950311',
    cardNumber: null,
    currencyCode: CurrencyCodes.EUR,
    amount: 1,
    purpose: 'other 123',
    transferDetails: null,
  },
  openapiData: {
    type: 'TO_CONTACT',
    toCard: null,
    toPhone: '79032950311',
    subType: null,
    externalOwnerID: null,
    fromCardPan: null,
  },
  operationType: null,
  fromPanDisplay: null,
};

export const transactionList = [transactionData, second];

export const transactionResponse: ITransactionResponse = {
  content: transactionList,
  maxPageNumber: 10,
  pageSize: 20,
  pageNumber: 1,
  length: 0,
};
