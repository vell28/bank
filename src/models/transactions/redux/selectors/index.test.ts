import { emptyStore } from 'modules/store/emptyStore';
import { isNotEmpty } from 'utils/ramda';
import { accountsData } from 'models/organizations/mock';

import { AccountTypes } from 'models/organizations/entities';
import { IStore } from 'modules/store/types';
import { CurrencyCodes } from 'modules/currencies';
import { transactionList } from '../../mock';
import {
  getShownBtns,
  getFilterDate,
  getTransactionsPageSize,
  getTransactionsPageNumber,
  getActiveFilter,
  getListLength,
  getReceiptTransaction,
  getTransactionsWithFilters,
} from '.';

const transactId = 'PPY184597';
const date = '2019-10-11T10:12:00';
const store: IStore = {
  ...emptyStore,
  organizations: {
    ...emptyStore.organizations,
    data: [
      {
        accounts: accountsData,
        cards: [],
        client: {},
        title: '',
        clientName: '',
        id: 0,
        name: '',
        tin: '',
      },
    ],
  },
  transactions: {
    isLoading: false,
    isSuccess: false,
    isError: false,
    data: {
      '11/10/2019': transactionList,
    },
    pagination: {
      maxPageNumber: 5,
      pageSize: 30,
      pageNumber: 1,
      length: 10,
    },
    filter: {
      date,
      accountType: AccountTypes.PHYSICAL,
      currency: CurrencyCodes.EUR,
      isShown: true,
    },
    settings: {
      shownBtns: { [transactId]: true },
      receiptId: transactId,
    },
    errors: [],
  },
};

describe('transactions selector tests', () => {
  test('getShownBtns should return ids of transaction', () => {
    const btns = getShownBtns(store);
    expect(btns[transactId]).toBeTruthy();
  });

  test('getFilterDate should return date or undefined if not set', () => {
    const selectorDate = getFilterDate(store);
    expect(selectorDate).toBe(date);
  });

  test('getTransactionsPageSize should return page size', () => {
    const size = getTransactionsPageSize(store);
    expect(size).toBe(30);
  });

  test('getTransactionsPageNumber should return page number', () => {
    const page = getTransactionsPageNumber(store);
    expect(page).toBe(1);
  });
  test('getActiveFilter should return active filter by type or ""', () => {
    const filter = getActiveFilter(store);
    expect(filter).toBe('PHYSICALEUR');
  });
  test('getListLength should return length of transactions', () => {
    const length = getListLength(store);
    expect(length).toBe(10);
  });
  test('getReceiptTransaction should return transaction by id(receiptId) ', () => {
    const transact = getReceiptTransaction(store);
    expect(transact).not.toBe(undefined);
    expect(transact).toMatchSnapshot();
  });
  test('getTransactionsWithFilters should return filtered transactions', () => {
    const transactions = getTransactionsWithFilters(store);
    expect(isNotEmpty(transactions)).toBeTruthy();
    expect(transactions).toMatchSnapshot();
  });
});
