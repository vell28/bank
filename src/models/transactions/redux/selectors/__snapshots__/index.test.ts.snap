// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transactions selector tests getReceiptTransaction should return transaction by id(receiptId)  1`] = `
Object {
  "accountId": "PPY5107",
  "amount": -1,
  "clientId": "6449",
  "currency": Object {
    "code": "EUR",
  },
  "description": "To: Elfur Long 12",
  "executedAt": "2019-10-11T10:08:47",
  "fee": 0,
  "fromBank": "Papaya",
  "fromName": "Ralf Williams",
  "fromNumber": "MT98PAPY36836000002676370004519",
  "fromPanDisplay": null,
  "id": "PPY184597",
  "openapiData": Object {
    "externalOwnerID": null,
    "fromCardPan": null,
    "subType": null,
    "toCard": null,
    "toPhone": "79032950311",
    "type": "TO_CONTACT",
  },
  "operationType": null,
  "paymentToRepeat": Object {
    "account": "PPY5107",
    "address": null,
    "amount": 1,
    "beneficiaryAccount": null,
    "beneficiaryBank": null,
    "beneficiaryName": null,
    "cardNumber": null,
    "city": null,
    "commissionType": null,
    "country": null,
    "currencyCode": "EUR",
    "fromCardId": null,
    "fromCardPan": null,
    "fromPhone": "79111111111",
    "iban": null,
    "intermediaryAccount": null,
    "intermediaryBank": null,
    "intermediarySwift": null,
    "phoneNumber": "79032950311",
    "purpose": "other 123",
    "swiftCode": null,
    "transferDetails": null,
    "type": "PAYMENT_PHONE",
    "urgency": null,
  },
  "paymentType": "TRANSFER",
  "referenceNumber": "PPY84f7087c-c511-4928-b299-13ee47c1bd9",
  "status": "COMPLETED",
  "toBank": "Papaya",
  "toName": "Elfur Long",
  "toNumber": "MT52PAPY36836000002676370000303",
}
`;

exports[`transactions selector tests getTransactionsWithFilters should return filtered transactions 1`] = `
Object {
  "11/10/2019": Array [
    Object {
      "accountId": "PPY5107",
      "amount": -10,
      "clientId": "6449",
      "currency": Object {
        "code": "EUR",
      },
      "description": "To: Dmitrii Popov Vvg",
      "executedAt": "2019-10-11T10:12:00",
      "fee": 0,
      "fromBank": "Papaya",
      "fromName": "Ralf Williams",
      "fromNumber": "MT98PAPY36836000002676370004519",
      "fromPanDisplay": null,
      "id": "PPY184598",
      "openapiData": Object {
        "externalOwnerID": null,
        "fromCardPan": null,
        "subType": null,
        "toCard": null,
        "toPhone": "79085491121",
        "type": "TO_CONTACT",
      },
      "operationType": null,
      "paymentToRepeat": Object {
        "account": "PPY5107",
        "address": null,
        "amount": 10,
        "beneficiaryAccount": null,
        "beneficiaryBank": null,
        "beneficiaryName": null,
        "cardNumber": null,
        "city": null,
        "commissionType": null,
        "country": null,
        "currencyCode": "EUR",
        "fromCardId": null,
        "fromCardPan": null,
        "fromPhone": "79111111111",
        "iban": null,
        "intermediaryAccount": null,
        "intermediaryBank": null,
        "intermediarySwift": null,
        "phoneNumber": "79085491121",
        "purpose": "other Vvgh",
        "swiftCode": null,
        "transferDetails": null,
        "type": "PAYMENT_PHONE",
        "urgency": null,
      },
      "paymentType": "TRANSFER",
      "referenceNumber": "PPY1e1d31e3-6acb-4d60-9f70-b9151221d11",
      "simpleAccount": Object {
        "accountName": "MT98PAPY36836000002676370004519",
        "accountType": "PHYSICAL",
        "accruedInterest": null,
        "availableBalance": 1897.61,
        "availableOperations": Array [],
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "bonusProgramActivate": false,
        "clientId": "6449",
        "color": "DEFAULT",
        "createdAt": "2019-11-24T14:59:55",
        "currency": Object {
          "code": "EUR",
        },
        "id": "PPY5107",
        "number": "MT98PAPY36836000002676370004519",
        "status": "ACTIVE",
      },
      "status": "COMPLETED",
      "toBank": "Papaya",
      "toName": "Dmitrii Popov",
      "toNumber": "MT97PAPY36836000002676370003470",
    },
    Object {
      "accountId": "PPY5107",
      "amount": -1,
      "clientId": "6449",
      "currency": Object {
        "code": "EUR",
      },
      "description": "To: Elfur Long 12",
      "executedAt": "2019-10-11T10:08:47",
      "fee": 0,
      "fromBank": "Papaya",
      "fromName": "Ralf Williams",
      "fromNumber": "MT98PAPY36836000002676370004519",
      "fromPanDisplay": null,
      "id": "PPY184597",
      "openapiData": Object {
        "externalOwnerID": null,
        "fromCardPan": null,
        "subType": null,
        "toCard": null,
        "toPhone": "79032950311",
        "type": "TO_CONTACT",
      },
      "operationType": null,
      "paymentToRepeat": Object {
        "account": "PPY5107",
        "address": null,
        "amount": 1,
        "beneficiaryAccount": null,
        "beneficiaryBank": null,
        "beneficiaryName": null,
        "cardNumber": null,
        "city": null,
        "commissionType": null,
        "country": null,
        "currencyCode": "EUR",
        "fromCardId": null,
        "fromCardPan": null,
        "fromPhone": "79111111111",
        "iban": null,
        "intermediaryAccount": null,
        "intermediaryBank": null,
        "intermediarySwift": null,
        "phoneNumber": "79032950311",
        "purpose": "other 123",
        "swiftCode": null,
        "transferDetails": null,
        "type": "PAYMENT_PHONE",
        "urgency": null,
      },
      "paymentType": "TRANSFER",
      "referenceNumber": "PPY84f7087c-c511-4928-b299-13ee47c1bd9",
      "simpleAccount": Object {
        "accountName": "MT98PAPY36836000002676370004519",
        "accountType": "PHYSICAL",
        "accruedInterest": null,
        "availableBalance": 1897.61,
        "availableOperations": Array [],
        "balance": 0,
        "balanceHold": 0,
        "balances": Array [
          Object {
            "availableBalance": 1897.61,
            "balance": 1030.05,
            "balanceHold": 0,
            "currency": "EUR",
          },
        ],
        "bonusProgramActivate": false,
        "clientId": "6449",
        "color": "DEFAULT",
        "createdAt": "2019-11-24T14:59:55",
        "currency": Object {
          "code": "EUR",
        },
        "id": "PPY5107",
        "number": "MT98PAPY36836000002676370004519",
        "status": "ACTIVE",
      },
      "status": "COMPLETED",
      "toBank": "Papaya",
      "toName": "Elfur Long",
      "toNumber": "MT52PAPY36836000002676370000303",
    },
  ],
}
`;
